modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -249,6 +271,42 @@ public GridMetricManager(GridKernalContext ctx) {
             spi.setMetricRegistry(this);
 
         startSpi();
+
+        ctx.internalSubscriptionProcessor().registerDistributedMetastorageListener(
```

Comment:
About metrics in metastorage: at the moment metastorage size will only grow because we never remove metrics related keys from metastorage. For example if user destroy cache than all key should be removed from metastore for this cache. The same for indexes for example and may be something else.

Reply:
One more case for hitrate and histogram configuration: during debug I want configure some metric only for one particular node. In this case we should keep this configuration in local metastorage. As result we also should change API and functionality because we should override local config during execution of distributed update. Also local key should override global key on node start. I think it could be done as separate issue.

Reply:
> During debug I want configure some metric only for one particular node

I thought we don't want to have local configuration? That's why the previous version of this PR was totally reworked.

Reply:
No, previous version of PR was reworked because it didn't provide distributed way for configuration. But local configuration still could be very helpful for debug when I can increase dimension of histogram for some node(s).

Reply:
OK. https://issues.apache.org/jira/browse/IGNITE-12426 created.